### PR TITLE
Feat/hamidreza rahmani

### PR DESCRIPTION
**Pull request by hamidreza-rahmani**
In this exercise I tried to use Semaphore for handle Concurrent requests and objectPooling

### CODE REVIEW

File: sources/hamidreza-rahmani/MicroWebFramework/ObjectPool.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+using System.Net;
+
+namespace MicroWebFramework;
+
+public class ObjectPool<T>
```
Comment (mohammadKarimi) [inline sources/hamidreza-rahmani/MicroWebFramework/ObjectPool.cs:5]: why you are using ObjectPool?

File: sources/hamidreza-rahmani/MicroWebFramework/Program.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+using System.Net;
+using MicroWebFramework;
+using MicroWebFramework.Middleware;
+using MicroWebFramework.pipeline;
+
+var pool = new ObjectPool<StateObject>(() => new StateObject(), 2);
+;
+var httpListener = new HttpListener();
+var semaphore = new SemaphoreSlim(10, 10);
```
Comment (mohammadKarimi) [inline sources/hamidreza-rahmani/MicroWebFramework/Program.cs:11]: needless complexity.

File: sources/hamidreza-rahmani/MicroWebFramework/pipeline/PiplineBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    public PiplineBuilder Add<TType>()
+    {
+        return Add(typeof(TType));
+    }
+
+    public Action<HttpContext> Build()
+    {
+        var currentPipe = (Pipe)Activator.CreateInstance(_pips[^1], null)!;
```
Comment (mohammadKarimi) [inline sources/hamidreza-rahmani/MicroWebFramework/pipeline/PiplineBuilder.cs:20]: range and index 👍

File: sources/hamidreza-rahmani/MicroWebFramework/pipeline/PiplineBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+{
+    private readonly List<Type> _pips = new();
+
+    public PiplineBuilder Add(Type pipe)
+    {
+        _pips.Add(pipe);
+        return this;
+    }
+
+    public PiplineBuilder Add<TType>()
```
Comment (mohammadKarimi) [inline sources/hamidreza-rahmani/MicroWebFramework/pipeline/PiplineBuilder.cs:13]: please filter TType with where keyword.

**Review by mohammadKarimi — CHANGES_REQUESTED** [review]
your code is perfect. but I think you struggle in overthinking and needless complexity. 
Be careful.

File: sources/hamidreza-rahmani/MicroWebFramework/ObjectPool.cs (hunk 1 above)
Comment (hamidreza-rahmani) [reply sources/hamidreza-rahmani/MicroWebFramework/ObjectPool.cs:5]: You are right I will try to fix this issue about needless complexity

**Review by mohammadKarimi — APPROVED** [review]

**Pull request merged**